### DISCUSSION TITLE
Getting started: Two-factor authentication

### DISCUSSION THREAD

**Discussion by tomslee** [SAP-archive/btp-setup-automator#213] [Q&A] [answered]
Thanks for creating this tool, which I am hoping will free up some of my own memory currently full of many different CLI commands.

I face a problem which is not your fault: my BTP account has two-factor authentication turned on, and for a number of reasons I would rather not turn it off. Unfortunately I have not even been able to run ./btpsa successfully. My attempt has been to type my >14-character  password (which includes spaces) and paste a 2-Factor Authentication code afterwards. All such attempts have failed so far.

I have tried setting loginmethod to "sso" in parameters.json, but the docker container says "Failed to open browser. Please authenticate manually". Any attempt to type or paste a 2FA code has so far not been successful (it just hangs with no response).

Any tips or suggestions or workarounds appreciated.

**Comment by lechnerc77** [comment, marked as answer]
Sorry for the delayed reply (vacation season :-) ). 
How do you access the btpsa CLI in Docker? Do you attach to a terminal only or do you attach via VSCode? Maybe you can give the VSCode approach a try as this should be capable of opening a browser window for your 2FA.

**Reply by lechnerc77** [reply]
No answer since > 1 month => mark as answered
